src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1825,6 +1855,18 @@ private void ExecuteGraphBuildScheduler(GraphBuildSubmission submission)
                     _overallBuildSuccess = false;
                 }
             }
+
+            static void DumpGraph(ProjectGraph graph, IReadOnlyDictionary<ProjectGraphNode, ImmutableList<string>> targetList = null)
+            {
+                if (Traits.Instance.DebugEngine is false)
```

Comment:
Why `is false`?

Also, nit: I'd negate this and move the rest in. Nice to have just one exit point for functions when it isn't too hard.

Reply:
I prefer to minimize the amount of statements in nested blocks. Function still has two exit points even if I put all the logic in the if statement.